src/cache-set/cache-set.ts

```diff
@@ -21,102 +59,158 @@ export class CacheSet<K extends string | number = string | number, V = any> exte
     const actualOptions: CacheSetOptions = options || (typeof iterableOrObject === 'object' ? iterableOrObject : { }) as CacheSetOptions;
 
     this.slots = actualOptions.slots || this.slots;
-    this.ttl = (actualOptions.ttl || this.ttl) * 1000;
+    this.ttl = actualOptions.ttl || this.ttl;
+    this.maxAge = actualOptions.maxAge || this.maxAge;
     this.keepPromises = actualOptions.keepPromises || this.keepPromises;
 
     if (iterable) {
       iterable.forEach(([k, v]) => this.set(k, v));
     }
   }
 
+  private updateUsages(key: K, push: 'delete'): undefined;
+  private updateUsages(key: K, push: 'read' | 'write'): number;
+  private updateUsages(key: K, type: 'read' | 'write' | 'delete'): undefined | number {
+    const index = this.usages.indexOf(key);
+
+    if (index !== -1) {
+      this.usages.splice(index, 1);
+    }
+
+    if (type !== 'delete') {
+      const now = Date.now();
+
+      this.usages.push(key);
+
+      if (type === 'write') this.lastWrite = now;
+
+      return this.lastUsage = now;
+    }
+  }
+
+  isAllExpired(): boolean {
+    if (this.size === 0) return false;
+
+    const now = Date.now();
+    const sinceLastUsage = now - this.lastUsage;
+    const sinceLastWrite = now - this.lastWrite;
+
+    return sinceLastUsage > this.ttl ||  sinceLastWrite > this.maxAge;
+  }
+
+  isItemExpired(key: K): boolean {
+    const item = super.get(key);
+
+    if (!item) return false;
+
+    const now = Date.now();
+    const sinceLastUsage = now - item.lastUsage;
+    const sinceLastWrite = now - item.createdAt;
+
+    return sinceLastUsage > this.ttl ||  sinceLastWrite > this.maxAge;
+  }
+
   clear() {
     super.clear();
 
     this.lastUsage = 0;
-    this.dates.clear();
+    this.lastWrite = 0;
     this.usages = [];
   }
 
   delete(key: K): boolean {
     const deleted = super.delete(key);
 
-    if (deleted) {
-      this.dates.delete(key);
-
-      const index = this.usages.indexOf(key);
-
-      if (index !== -1) {
-        this.usages.splice(index, 1);
-      }
-    }
+    if (deleted) this.updateUsages(key, 'delete');
 
     return deleted;
   }
 
-  get(key: K): V | undefined {
+  get(key: K): CacheItem<K, V> | undefined {
     if (this.has(key)) {
-      this.dates.set(key, this.lastUsage = Date.now());
-      this.usages.splice(this.usages.indexOf(key), 1);
-      this.usages.push(key);
+      const lastUsage = this.updateUsages(key, 'read');
 
-      return super.get(key);
+      const nextItem: CacheItem<K, V> = {
+        ...super.get(key)!,
+        lastUsage,
+      };
+
+      super.set(key, nextItem);
+
+      return nextItem;
     }
   }
 
-  has(key: K): boolean {
-    const now = Date.now();
+  getValue(key: K): V | undefined {
```

Comment:
I'll probably use composition instead of extending from Map and have the `get` return the value and have a separated one called `getItem` or `getEntry` to get the whole cache item object.